rclcpp/src/rclcpp/context.cpp

```diff
@@ -162,6 +162,9 @@ Context::shutdown(const std::string & reason)
       ++it;
     }
   }
+  // clear sub contexts
+  std::lock_guard<std::recursive_mutex> lock(sub_contexts_mutex_);
+  sub_contexts_.clear();
```

Comment:
I don't think this is a safe solution to this issue because things may re-add the sub contexts after shutdown. See my comment:

https://github.com/ros2/rclcpp/issues/893#issuecomment-555170318

Reply:
@wjwwood 

Thanks for your comments.   
Based on your comments https://github.com/ros2/rclcpp/issues/893#issuecomment-555170318, I will use weak pointer to fix this issue